src/signature.c

```diff
@@ -2009,3 +2049,274 @@ const OSSL_DISPATCH p11prov_ecdsa_signature_functions[] = {
     DISPATCH_SIG_ELEM(ecdsa, SETTABLE_CTX_PARAMS, settable_ctx_params),
     { 0, NULL },
 };
+
+DISPATCH_EDDSA_FN(newctx);
+DISPATCH_EDDSA_FN(digest_sign_init);
+DISPATCH_EDDSA_FN(digest_sign);
+DISPATCH_EDDSA_FN(digest_verify_init);
+DISPATCH_EDDSA_FN(digest_verify);
+DISPATCH_EDDSA_FN(get_ctx_params);
+DISPATCH_EDDSA_FN(set_ctx_params);
+DISPATCH_EDDSA_FN(gettable_ctx_params);
+DISPATCH_EDDSA_FN(settable_ctx_params);
+
+static void *p11prov_eddsa_newctx(void *provctx, const char *properties)
+{
+    P11PROV_CTX *ctx = (P11PROV_CTX *)provctx;
+    P11PROV_SIG_CTX *sigctx;
+
+    sigctx = p11prov_sig_newctx(ctx, CKM_EDDSA, properties);
+    if (sigctx == NULL) {
+        return NULL;
+    }
+
+    return sigctx;
```

Comment:
nit: Both of the branches do the same so this could be simplified to a single line:
```suggestion

    return p11prov_sig_newctx(ctx, CKM_EDDSA, properties);
```
unless we would want to do some more stuff here, keep the same structure as with other mechanisms or do some logging on error (OOM warning or something).

Reply:
Initially the function was more complicated, then it reduced to what is now, no need to keep it complicated. Fixing.